client_api.go

```diff
@@ -158,13 +172,31 @@ func (c *Client) CertificateRevoke(
 	ctx context.Context,
 	serial *big.Int,
 ) error {
+	return c.CertificateRevokeWithReason(ctx, serial, RevocationReasonUnspecified)
+}
+
+// CertificateRevokeWithReason revokes a certificate with a specified reason.
+func (c *Client) CertificateRevokeWithReason(
+	ctx context.Context,
+	serial *big.Int,
+	reason RevocationReason,
+) error {
+	type certificatePatch struct {
+		RevocationReason RevocationReason `json:"revocation_reason"`
+	}
+
+	var patch = certificatePatch{
+		RevocationReason: RevocationReasonUnspecified,
```

Comment:
should we provide here the reason from the parameter passed to this method ?

Reply:
Yup!